packages/solid-query/src/useMutationState.ts

```diff
@@ -0,0 +1,61 @@
+import { createMemo, createSignal, onCleanup } from 'solid-js'
+import { replaceEqualDeep } from '@tanstack/query-core'
+import { useQueryClient } from './QueryClientProvider'
+import type {
+  DefaultError,
+  Mutation,
+  MutationCache,
+  MutationFilters,
+  MutationState,
+} from '@tanstack/query-core'
+import type { Accessor } from 'solid-js'
+import type { QueryClient } from './QueryClient'
+
+type MutationStateOptions<TResult = MutationState> = {
+  filters?: MutationFilters
+  select?: (
+    mutation: Mutation<unknown, DefaultError, unknown, unknown>,
+  ) => TResult
+}
+
+function getResult<TResult = MutationState>(
+  mutationCache: MutationCache,
+  options: MutationStateOptions<TResult>,
+): Array<TResult> {
+  return mutationCache
+    .findAll(options.filters)
+    .map(
+      (mutation): TResult =>
+        (options.select
+          ? options.select(
+              mutation as Mutation<unknown, DefaultError, unknown, unknown>,
+            )
+          : mutation.state) as TResult,
+    )
+}
+
+export function useMutationState<TResult = MutationState>(
+  options: Accessor<MutationStateOptions<TResult>> = () => ({}),
+  queryClient?: Accessor<QueryClient>,
+): Accessor<Array<TResult>> {
+  const client = createMemo(() => useQueryClient(queryClient?.()))
+  const mutationCache = createMemo(() => client().getMutationCache())
+
+  const [result, setResult] = createSignal(
+    getResult(mutationCache(), options()),
+  )
+
+  const unsubscribe = mutationCache().subscribe(() => {
+    const nextResult = replaceEqualDeep(
+      result(),
+      getResult(mutationCache(), options()),
+    )
+    if (result() !== nextResult) {
+      setResult(nextResult)
+    }
+  })
+
+  onCleanup(unsubscribe)
```

Comment:
Last fix I swear Can we wrap this in a `createEffect`?

```ts
createEffect(() => {
    const unsubscribe = mutationCache().subscribe(() => {
    const nextResult = replaceEqualDeep(
      result(),
      getResult(mutationCache(), options()),
    )
    if (result() !== nextResult) {
      setResult(nextResult)
    }
  })

  onCleanup(unsubscribe)
})
```

The reason to do this is that if the `options` object changes in the client code. We should run the sub function with the new `options`

For example:
```ts
// First render
const [count, setCount] = createSignal(0)

const mutationState = useMutationState(() => ({
  mutationKey: ['mutation', count()] // count() is 0
}))
```

```ts
// When count is updated to 2
const [count, setCount] = createSignal(0)

const mutationState = useMutationState(() => ({
  mutationKey: ['mutation', count()] // count() is 2
}))

// This will work now!!!
```

Reply:
@brenelz  last change and we're good to go on this PR

Reply:
Yeah thats a good catch. Should be good now